### ISSUE TITLE
Ordering grid columns based on the order of ngFor array

### ISSUE THREAD

**Issue by baleeds** [IgniteUI/igniteui-angular#11493]
## Description  
When changing the order of columns in the view, using an *ngFor directive, the columns in the grid itself aren't reordered.

 * igniteui-angular version: 12.3.6
 * browser: Chrome

## Steps to reproduce  
See this stackblitz:
https://stackblitz.com/edit/angular-bejvgc-pxqlom

Clicking "reverse" will reverse the column order, and you can see the changes in the field display, but they aren't seen in the grid.

## Result  
Columns don't move.

## Expected result  
Columns move based on the updated array order.

**Comment by hanastasov** [comment]
I've tried the `updateColumns` method (see https://stackblitz.com/edit/angular-bejvgc-nsvgjz?file=src/app/grid/grid-sample-pinning/grid-toolbar-pinning.component.ts) and still no difference. @Volen99 , could you please have a look at that one?

**Comment by Volen99** [comment]
Hello, @baleeds!

Thank you for taking your time to share this issue with us :)

After investigating, I found that the problem is in the sample in method **gridResetColumns**. After minor changes I believe everything is working now, please check [this ](https://stackblitz.com/edit/angular-bejvgc-s94wlv?file=src%2Fapp%2Fgrid%2Fgrid-sample-pinning%2Fgrid-toolbar-pinning.component.ts)stackblitz :)

**Comment by baleeds** [comment]
To clarify, we should `updateColumns` to an empty array?  It's an interesting workaround.  
```ts
private gridResetColumns(columns: any[]) {
    const newColumns = [];


    this.grid1.updateColumns(newColumns);
    this.grid1.cdr.detectChanges();
  }
```

Our project is running v12 right now, is there an equivalent function to `updateColumn`?

**Comment by Volen99** [comment]
Hey again, @baleeds.

Please excuse me, I failed to notice that the version you are targeting is 12.3.6 and there indeed is missing `updateColumns` method. 

This method is not working as expected and a separate [issue ](https://github.com/IgniteUI/igniteui-angular/issues/11575)has been logged.

equivalent method to `updateColumns` is  `reset` aka `this.grid1.columnList.reset(newColumns);`

I am still working on how to fix `updateColumns `method and to make `reset` method to work with `this.columns`

I will let you know once a solution is found :)

**Comment by baleeds** [comment]
@Volen99 do you have a timeline on when you might find the solution?

**Comment by hanastasov** [comment]
> @Volen99 thanks again for looking into this. Do you have a timeline on when you might find the solution?

We've planned this issue for next sprint, starting on Monday. This means a patch would be released most probably the week after the next one.

**Comment by hanastasov** [comment]
Ignore the suggestion to use `updateColumns`. This is actually not part of the public API, it requires a slightly different usage then what we tried above and we cannot guarantee that if method works one way this version, it will work the same afterwards.

Actually, there is no means in the public API to facilitate the required. 

With the above being said, you have taken the correct approach to modify the columns collection which columns are bound to. We continue investigating why in this approach columns are not affected.

**Comment by hanastasov** [comment]
@amitmoondra this was my expectation but turned out we need more time for the fix.

**Comment by amitmoondra** [comment]
@hanastasov thank you very much.

**Comment by amitmoondra** [comment]
@hanastasov will this be released for version 13.1.X?

**Comment by amitmoondra** [comment]
understood, I was planning to do 13.2.X but as i was using 13.1.X just wanted to ask.
